internal/querynode/segment_loader.go

```diff
@@ -328,14 +335,64 @@ func (loader *segmentLoader) loadFiledBinlogData(segment *Segment, fieldBinlogs
 		}
 
 		return loader.loadGrowingSegments(segment, rowIDData.(*storage.Int64FieldData).Data, utss, insertData)
-	case segmentTypeSealed:
-		return loader.loadSealedSegments(segment, insertData)
+
 	default:
-		err := errors.New(fmt.Sprintln("illegal segment type when load segment, collectionID = ", segment.collectionID))
+		err := fmt.Errorf("illegal segmentType=%v when load segment, collectionID=%v", segmentType, segment.collectionID)
 		return err
 	}
 }
 
+func (loader *segmentLoader) loadSealedSegmentFields(segment *Segment, fields []*datapb.FieldBinlog) error {
+	// Load fields concurrently
+	futures := make([]*concurrency.Future, 0, len(fields))
+	for _, field := range fields {
+		future := loader.loadSealedFieldAsync(segment, field)
+
+		futures = append(futures, future)
+	}
+
+	err := concurrency.AwaitAll(futures...)
+	if err != nil {
+		return err
+	}
+
+	log.Info("log field binlogs done",
+		zap.Int64("collection", segment.collectionID),
+		zap.Int64("segment", segment.segmentID),
+		zap.Any("fields", fields))
+
+	return nil
+}
+
+// async load field of sealed segment
+func (loader *segmentLoader) loadSealedFieldAsync(segment *Segment, field *datapb.FieldBinlog) *concurrency.Future {
+	iCodec := storage.InsertCodec{}
+
+	// Avoid consuming too much memory if no CPU worker ready,
+	// acquire a CPU worker before load field binlogs
+	return loader.cpuPool.Submit(func() (interface{}, error) {
+		futures := loader.loadFieldBinlogsAsync(field)
```

Comment:
should move the IO part out of cpuPool?

Reply:
That may lead to accumulate many binlogs in memory